parallel.go

```diff
@@ -289,20 +289,59 @@ func (r Parallel) forKey(key string) Parallel {
 	})
 }
 
+// mergeQueryEvents limits `routing.QueryError` events to only be sent on the context in case all parallel
+// routers fail.
+func (r Parallel) mergeQueryEvents(ctx context.Context) (context.Context, context.CancelFunc) {
+	subCtx, evCh := routing.RegisterForQueryEvents(ctx)
+	subCtx, cancel := context.WithCancel(subCtx)
```

Comment:
I believe:

1. This needs to go one line up so we actually cancel the query events context.
2. Below, we should _not_ select on this context so we read the last query event.

Right?